### ISSUE TITLE
Getting custom oauthData from provider

### ISSUE THREAD

**Issue by rothlis** [sahat/satellizer#237]
One of the provider I'm working with (i.e. Vend POS) is sending `domain_prefix` alongside the `code` and the rest of the OAuth data.

Are there any solutions for the server to receive that param (or any others) as well?

**Comment by sahat** [comment]
At the moment you cannot because it is only set to get `code`:
```js
          oauth2.exchangeForToken = function(oauthData, userData) {

            var data = angular.extend({}, userData, {
              code: oauthData.code,
              clientId: defaults.clientId,
              redirectUri: defaults.redirectUri
            });

            return $http.post(defaults.url, data, {withCredentials: true});
          };
```

I will need to think about how to make it more flexible without breaking existing functionality for other users.

**Comment by sahat** [comment]
Here is what I came up with:

```js
 $authProvider.oauth2({
      name: 'vend_pos',
      url: '/auth/vend_pos',
      clientId: 'your client id',
      responseParams: ['domain_prefix'],
      redirectUri: window.location.origin || window.location.protocol + '//' + window.location.host,
      authorizationEndpoint: 'authenticate url'
});
```

And this is what you will get on the server, more or less:

```js
{code: "AQD7z6rypQJqvm-H4qRp", clientId: "2318318231231", redirectUri: "http://localhost:3000/", domain_prefix: "your domain prefix"}
```

If this is what you had in mind, I will make a patch for it. Also, what about the name `responseParams`?

Here is the full list of OAuth2 params now:

```js
var defaults = {
            url: null,
            name: null,
            scope: null,
            scopeDelimiter: null,
            clientId: null,
            redirectUri: null,
            popupOptions: null,
            authorizationEndpoint: null,
            responseParams: null,
            requiredUrlParams: null,
            optionalUrlParams: null,
            defaultUrlParams: ['response_type', 'client_id', 'redirect_uri'],
            responseType: 'code'
};
```

**Comment by rothlis** [comment]
Hi @sahat, thanks for the quick response!

This is exactly what I had in mind and would help us a lot. The name `responseParams` or even `callbackParams` works for me.

As for the options `requiredUrlParams` and `optionalUrlParams` and looking through the code, they don't differ functionally. So yes, it might be a good idea to merge them in something like `urlParams` or `requestParams`.

Seb

**Comment by sahat** [comment]
Available in 0.9.0.

**Comment by rothlis** [comment]
@sahat Thanks for the great work!
